client/packages/system/src/Vaccination/api/useVaccination.ts

```diff
@@ -217,46 +221,58 @@ const useInsert = ({
   });
 };
 
-const useUpdate = (vaccinationId: string | undefined) => {
+const useUpdate = (vaccination: VaccinationQuery['vaccination']) => {
   const { api, storeId, queryClient } = useVaccinationsGraphQL();
   const t = useTranslation();
 
   const mutationFn = async (input: VaccinationDraft) => {
-    if (!vaccinationId) {
+    if (!vaccination) {
       throw new Error(t('error.failed-to-save-vaccination'));
     }
 
     const isOtherFacility = input.facilityId === OTHER_FACILITY;
 
+    // We should send a reduced input to the API if vaccination was given
+    // from another store
```

Comment:
don't love this solution, in that if we update inputs in the future, might be easy to forget? But I think ok for now...